### ISSUE TITLE
Assign default literal to a deconstruction

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#19513]
**Version Used**: master

**Steps to Reproduce**:

```cs
class C {
    void M(int i , int j) {
        (i, j) = default;
    }
}
```
**Expected Behavior**: compile with no error.

**Actual Behavior**: error CS8131: Deconstruct assignment requires an expression with a type on the right-hand-side.

**Comment by alrz** [comment]
@jcouv Can this be special cased? [would come in handy sometime](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs,07114598e8759e4e).

**Comment by jcouv** [comment]
Issue moved to [dotnet/csharplang #583](https://github.com/dotnet/csharplang/issues/583) via [**ZenHub**](https://www.zenhub.com/)
